SWIPs/swip-streaming-of-feeds.md

```diff
@@ -0,0 +1,148 @@
+This proposed feed indexing method is the opposite as at lookups: 
+if there is expected amount of writes periodically, the retrieval of the data is faster, that even can be _O(1)_. 
+Compared to the epoch-based feeds, basically the length of the base segment is arbitrary, and it encourages the user to make periodic uploads and stick to this base segment instead of sporadic uploading in exchange for better retrieval time.
+
+The downside is within the concerted update time period the content creator cannot update the state of the feed - although it is possible to overwrite the content of a feed by uploading it again with different payload, but on download-side it is possible to still to get back the old one. 
+This problem can be mitigated by changing the registry of the feed, from where the consumers get the initial metadatas of the feed: 
+- the initial timestamp (`T0`) should be changed to that point from where the time period changes (`T1`)
+- change the new time period (`Δ1`) from the old one (`Δ0`) for state uploads
+
+If the registry settled on blockchain in a smart contract, then it can have a defined event on metadata change of the content, on which the clients can listen.
+Thereby, the consumers of the feed can immediately react to the upload frequence change and they can poll according to the new rules.
+If somehow the blockchain listening on client side is not suitable, it is possible to put the `uploading time period` and `initial timestamp` metadatas into all of the feed stream updates so that the consumers can sync to the stream after `MAX(Δ0,Δ1)` time if `MAX(Δ0,Δ1) mod MIN(Δ0,Δ1) = 0` and  there is one `k` and `m` positive integer where `T0 + (k * Δ0) = T0 + (m * Δ1) = T1`
+
+Though it is stated this approach does not address downloading the whole feed stream, it is still possible:
+* In case of punctual updates without any upload time period change, the stream download is identical with the sequential/periodic feed stream download.
+* If the uploading time period has been changed, the feed index set construction should happen either considering the before mentioned blockchain event emitting or if it is contained only in the feed metadata, then when it changes in the downloading process it is necessary to start a lookup backwards if `Δ1 < Δ0` until the change, but maximum `z = Δ0 / Δ1` units.
+
+## Backwards Compatibility
+<!--All SWIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The SWIP must explain how the author proposes to deal with these incompatibilities. SWIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+The whole idea can be implemented on application layer using single owner chunks, but optionally the solution also can be integrated to the P2P client.
```

Comment:
we would need  every publisher update to emit an event (PSS) and a version/height eg height=1 + feed topic gets the 1st update

Reply:
it is not related to PSS